W-15707735-Add variable issue to Java doc (#246)
mulesoft/docs-general: modules/ROOT/pages/java-support.adoc

[[dataweave]]
== DataWeave

DataWeave uses Java's reflection API to read and write Java objects. Java 17 adds some restrictions in encapsulation and reflective access that affect the Java Data Format and Mule objects.

To ensure that your applications continue to work as expected, follow these guidelines:

* Verify that the objects used by your application are Plain Old Java Objects (POJOs). 
+
POJOs are required for DataWeave 2.6.0 and later, and you must ensure that the POJOs have:

** A default constructor
** Getters for all properties
** Setters for all properties

> * Verify that predefined variables are accessed through their proper API

Reviewer: ```suggestion
* Verify that predefined variables are accessed through their proper API.
```

+

> Variables like `attributes`, `message`, or `error` have specific access APIs. Using the proper fields is required when using Java 17 because internal fields are no longer accesible. For example:

Reviewer: ```suggestion
Variables like `attributes`, `message`, or `error` have specific access APIs. Using the proper fields is required when using Java 17 because internal fields are no longer accessible. For example:
```

> Variables like `attributes`, `message`, or `error` have specific access APIs. Using the proper fields is required when using Java 17 because internal fields are no longer accessible. For example:

Reviewer: ```suggestion
Variables like `attributes`, `message`, and `error` have specific access APIs. Using the proper fields is required when using Java 17 because internal fields are no longer accessible. For example:
```

> 

Reviewer 2: Would "and" be better than "or" here?

Author: Not sure, I was trying to be vague because there are most variables that apply. With "and" the list seems more definitive.

> ** `error.muleMessage` will need to be replaced by the proper Mule error field `error.errorMessage`

Reviewer: ```suggestion
** `error.muleMessage` needs to be replaced by the proper Mule error field `error.errorMessage`.
```

> ** `error.errors` will need to be replaced by the proper Mule error field `error.childErrors`

Reviewer: ```suggestion
** `error.errors` needs to be replaced by the proper Mule error field `error.childErrors`.
```
